homeassistant/components/honeygain/sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Sensors for HoneyGain data."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CURRENCY_DOLLAR
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HoneygainData
+from .const import DOMAIN
+
+
+@dataclass
+class SensorValueEntityDescription(SensorEntityDescription):
+    """Class describing Honeygain sensor entities."""
+
+    value: Callable = lambda x: x
+
+
+HONEYGAIN_SENSORS: list[SensorValueEntityDescription] = [
+    SensorValueEntityDescription(
+        key="account_balance",
+        name="Account Balance",
+        icon="mdi:hand-coin",
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement=CURRENCY_DOLLAR,
+        value=lambda x: x.balances.get("payout").get("usd_cents") / 100,
+    ),
+    SensorValueEntityDescription(
+        key="today_earnings",
+        name="Today's Earnings",
+        icon="mdi:calendar-today",
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement=CURRENCY_DOLLAR,
+        value=lambda x: x.balances.get("realtime").get("usd_cents") / 100,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Sensor set up for HoneyGain."""
+    honeygain_data: HoneygainData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[SensorEntity] = [
+        HoneygainAccountSensor(honeygain_data, sensor_description)
+        for sensor_description in HONEYGAIN_SENSORS
+    ]
+    async_add_entities(entities)
+
+
+class HoneygainAccountSensor(SensorEntity):
+    """Sensor to track Honeygain data."""
+
+    honeygain_data: HoneygainData
+    entity_description: SensorValueEntityDescription
+
+    def __init__(
+        self,
+        honeygain_data: HoneygainData,
+        sensor_description: SensorValueEntityDescription,
+    ) -> None:
+        """Create Sensor for displaying Honeygain account details."""
+        self.entity_description = sensor_description
+        self._honeygain_data = honeygain_data
+        self.entity_id = f"sensor.{sensor_description.key}"
+        self._attr_name = f"Honeygain {sensor_description.name}"
```

Comment:
I think you do not have to set this if `self.entity_description` is set

Reply:
I was trying to set pleasing `entity_id` s and `_name` s but we can just let the sensor create those